alternator/executor.cc

```diff
@@ -159,31 +156,41 @@ static std::string lsi_name(const std::string& table_name, const std::string& in
  *  and api_error in case the table name is missing or not a string, or
  *  doesn't pass validate_table_name().
  */
-static std::string get_table_name(const rjson::value& request) {
-    const rjson::value& table_name_value = rjson::get(request, "TableName");
-    if (!table_name_value.IsString()) {
+std::optional<std::string> executor::get_table_name(const rjson::value& request, bool required) {
```

Comment:
Maybe. I tried to keep each change as small and isolated as possible. You could try reviewing in the "changed files" view instead of the individual patches (at least first). Gives a more end-goal view.